service/create/keypair.go

```diff
@@ -0,0 +1,63 @@
+package create
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	microerror "github.com/giantswarm/microkit/error"
+
+	awsclient "github.com/giantswarm/aws-operator/client/aws"
+)
+
+type keyPairProvider interface {
+	pubKeyContent() ([]byte, error)
+}
+
+type fsKeyPairProvider struct {
+	pubKeyFile string
+}
+
+func newFsKeyPairProvider(pubKeyFile string) *fsKeyPairProvider {
+	return &fsKeyPairProvider{
+		pubKeyFile: pubKeyFile,
+	}
+}
+
+func (f *fsKeyPairProvider) pubKeyContent() ([]byte, error) {
+	return ioutil.ReadFile(f.pubKeyFile)
+}
+
+type keyPairInput struct {
+	ec2Client   *ec2.EC2
+	clusterName string
+	provider    keyPairProvider
+}
+
+func (s *Service) keyPair(input keyPairInput) (string, error) {
+	pkc, err := input.provider.pubKeyContent()
+	if err != nil {
+		return "", microerror.MaskAny(err)
+	}
+
+	keyPair, err := input.ec2Client.ImportKeyPair(&ec2.ImportKeyPairInput{
+		KeyName:           aws.String(input.clusterName),
+		PublicKeyMaterial: pkc,
+	})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == awsclient.KeyPairDuplicate {
+			s.logger.Log("info", fmt.Sprintf("keypair '%s' exists, reusing", input.clusterName))
+			return input.clusterName, nil
+		} else {
+			return "", microerror.MaskAny(err)
+		}
+	}
+
+	if keyPair != nil && keyPair.KeyName != nil {
+		return *keyPair.KeyName, nil
+	}
+
+	return "", fmt.Errorf("Couln't create and find the keypair '%s'", input.clusterName)
```

Comment:
I mean that in other parts of the code, we end with the positive outcome, and have handle the error cases in `ifs`, instead of the other way around, cfr https://github.com/kinvolk/aws-operator/blob/d26f43fb71bf546aaab558e7f83d08dedf8e7508/service/create/service.go#L447-L453 or https://github.com/kinvolk/aws-operator/blob/d26f43fb71bf546aaab558e7f83d08dedf8e7508/service/create/policy.go#L78-L82